### ISSUE TITLE
app.htmlkit.localization.set is getting overridden 

### ISSUE THREAD

**Issue by dev-onepayclick** [vapor-community/HTMLKit#124]
Hi,

Even tho i'm setting the locale using `app.htmlkit.localization.set(locale: "pt")` I keep getting the error **.missingTables** because the **currentLocale** keeps defaulting to my browser's locale "en".

Not sure it's a bug or I'm doing something wrong 😄

**Comment by mattesmohr** [comment]
Hey,

I can take a look tonight, when I am back home from work. I guess you are using the alpha? 

Or wich release version do you use exactly?

**Comment by mattesmohr** [comment]
@dev-onepayclick Did you set the source directory too? 

```swift
app.htmlkit.localization.set(source: currentDirectory)
```

How does your folder structure look like?

```
| Localization
| - en-GB
| - - translations.strings
```

**Comment by dev-onepayclick** [comment]
Hey, sorry for the late response. 

I fixed the folder structure. But then I get a different error. MissingKey. 
I wanted to force the use of 
```
| Localization
| - pt-BR
| - - translations.strings
```

so I called `app.htmlkit.localization.set(locale: "pt-BR")` but the library still defaults to en-CA.

Also, the error message is MissingKey but it doesn't tell the key that is missing.

**Comment by mattesmohr** [comment]
No problem. 👍 Can you give me an example, what your translations.strings is build up, so I can reproduce your issue. That would be great.

I think the second issue is on me. You can rename the folder `pt-BR` to `pt` and use `app.htmlkit.localization.set(locale: "pt")` for now. The localization does not know the Locale pt-BR yet.

I will fix it.

**Comment by dev-onepayclick** [comment]
Also, is there any way to opt out of localization?

**Comment by mattesmohr** [comment]
I am not sure I understand. You don't need to use the localization?

Can you expand your thought a bit more? What would you like to do?

**Comment by dev-onepayclick** [comment]
This is my current setup
![image](https://github.com/vapor-community/HTMLKit/assets/121349833/eaa1f1d0-db0c-46f0-938a-61e3d8cce09a)

![image](https://github.com/vapor-community/HTMLKit/assets/121349833/61970ef0-12eb-47b4-b845-848b3c8654ff)

![image](https://github.com/vapor-community/HTMLKit/assets/121349833/40cbf4d8-1bb5-4c94-a168-63b18fa7cf69)

But it seems it is still defaulting to en-CA. I changed the folder to "pt" as you suggested but it still goes uses "en-CA"

**Comment by dev-onepayclick** [comment]
>I am not sure I understand. You don't need to use the localization.
oh sorry. I was just curious, in case I needed to opt out of a translation for some specific case.
But I figured I can do
```
Label { "Something"}
```
in case I need to avoid localizing it.


and if I want it localized I do `Label("Something"`

**Comment by mattesmohr** [comment]
The language and region on your device is English Canadian, right? I wrote the localization to take the device settings at first. But I guess it is too prominent... Maybe it needs a change. I will rethink it.

In your case it uses Portuguese only as fallback, you know. Is it what you want? Or should Portuguese be your first language of choice?

By the way, I think `app.htmlkit.localization.set(source: URL(fileURLWithPath: "Localization"))` is not right, does it work? It should be "Sources/`Your Projectname`/Localization/". At least on a Mac system.

Back to the Portuguese locale. 

The current order for the localization is environment locale > device setting >  application setting (alias fallback). That said...

1. `import HTMLKit`
2. `application.htmlkit.environment.locale = Locale(tag: .portuguese)`

would force the localization to use Portuguese.

or

```swift
Html {
}
.environment(key: \.locale, value: Locale(tag: .portuguese))
```

```swift
Paragraph("hello")
        .environment(key: \.locale)
```

should do the same.

**Comment by mattesmohr** [comment]
> in case I need to avoid localizing it.
> 
> and if I want it localized I do `Label("Something"`

Thats exactly how it works right now.

**Comment by dev-onepayclick** [comment]
>By the way, I think app.htmlkit.localization.set(source: URL(fileURLWithPath: "Localization")) is not right, does it work? It should be "Sources/Your Projectname/Localization/". At least on a Mac system.

Yep, it seems to work. Its running on Docker tho. Haven't tried on Mac.

Yes I'm writing the webpage for a Portuguese audience :) 

Thanks for the tips 🙏🏻 I totally forgot about the .environment thing.

**Comment by mattesmohr** [comment]
You were absolutely right. It was a bug back then. I fixed it with #163 and released it with alpha.10.
